modules/collect/src/main/java/com/opengamma/strata/collect/io/XmlFile.java

```diff
@@ -101,6 +102,35 @@ public static XmlFile of(ByteSource source, String refAttrName) {
     });
   }
 
+  /**
+   * Parses the element names and structure from the specified XML, filtering to reduce memory usage.
+   * <p>
+   * This parses the specified byte source expecting an XML file format.
+   * The filter function should return the number of levels below the current one
+   * that should be included in the response.
+   * Always returning {@code Integer.MAX_VALUE} will not filter the children.
```

Comment:
So the range of `Integer.MIN_VALUE <= x < 0` is invalid values? Talking of which `Integer.MIN_VALUE` could cause interesting behaviour underflowing back to MAX_VALUE, which would remain for all children, and have the same effect as MAX_VALUE!